examples/vision/randaugment.py

```diff
@@ -0,0 +1,296 @@
+            layers.MaxPooling2D(pool_size=(2, 2)),
+            layers.Conv2D(32, (5, 5), activation="relu"),
+            layers.MaxPooling2D(pool_size=(2, 2)),
+            layers.Dropout(0.2),
+            layers.GlobalAvgPool2D(),
+            layers.Dense(128, activation="relu"),
+            layers.Dense(10, activation="softmax"),
+        ]
+    )
+    return model
+
+
+"""
+We will train this network on two different versions of our dataset:
+* One augmented with RandAugment.
+* Another one augmented with `simple_aug`.
+
+Since RandAugment is known to enhance the robustness of models to common perturbations
+and corruptions, we will also evaluate our models on the CIFAR-10-C dataset, proposed in
+[Benchmarking Neural Network Robustness to Common Corruptions and
+Perturbations](https://arxiv.org/abs/1903.12261) by Hendrycks et al. CIFAR-10-C dataset
+consists of 19 different image corruptions and perturbations (for example speckle noise,
+fog, Gaussian blur, etc.) that too at varying severity levels. For this example we will
+be using the following configuration:
+[`cifar10_corrupted/saturate_5`](https://www.tensorflow.org/datasets/catalog/cifar10_corrupted#cifar10_corruptedsaturate_5). The images from this configuration look like so:
+
+![](https://storage.googleapis.com/tfds-data/visualization/fig/cifar10_corrupted-saturate_5-1.0.0.png)
+
+For the sake of reproducibility, we serialize the initial random weights of our shallow
+network.
+"""
+
+initial_model = get_training_model()
+initial_model.save_weights("initial_weights.h5")
+
+"""
+#1. Train model with RandAugment
+"""
+
+rand_aug_model = get_training_model()
+rand_aug_model.load_weights("initial_weights.h5")
+rand_aug_model.compile(
+    loss="sparse_categorical_crossentropy", optimizer="adam", metrics=["accuracy"]
+)
+rand_aug_model.fit(train_ds_rand, validation_data=test_ds, epochs=EPOCHS)
+_, test_acc = rand_aug_model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_acc * 100))
+
+"""
+# 2. Train model with `simple_aug`
+"""
+
+simple_aug_model = get_training_model()
+simple_aug_model.load_weights("initial_weights.h5")
+simple_aug_model.compile(
+    loss="sparse_categorical_crossentropy", optimizer="adam", metrics=["accuracy"]
+)
+simple_aug_model.fit(train_ds_simple, validation_data=test_ds, epochs=EPOCHS)
+_, test_acc = simple_aug_model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_acc * 100))
+
+"""
+## Load the CIFAR-10-C dataset and evaluate performance
+"""
+
+# Load and prepare the CIFAR-10-C dataset
+cifar_10_c = tfds.load("cifar10_corrupted/saturate_5", split="test", as_supervised=True)
+cifar_10_c = cifar_10_c.batch(BATCH_SIZE)
+
+# Evaluate `rand_aug_model`
+_, test_acc = rand_aug_model.evaluate(cifar_10_c, verbose=0)
+print(
+    "Accuracy with RandAugment on CIFAR-10-C (saturate_5): {:.2f}%".format(
+        test_acc * 100
+    )
+)
+
+# Evaluate `simple_aug_model`
+_, test_acc = simple_aug_model.evaluate(cifar_10_c, verbose=0)
+print(
```

Comment:
What results do you get after training to convergence? Is 10 epochs enough? Don't worry about the notebook taking too much time -- we can run the experiment at full scale, then put an artificial number of epochs in the code (like 1) while clearly explaining how to reproduce the actual results

Reply:
Results with RandAugment are far better. Here's the [Notebook](https://colab.research.google.com/drive/1IYvdqGpJBfq-7nK_QDXCCBgEK-2_O1C5#scrollTo=c3j-pPzjdk8P). But yes, the models were not trained to completion. I agree with your suggestion on running the experiments at full scale. Will do it.